includes/rest-api/class-sensei-rest-api-data-port-controller.php

```diff
@@ -154,6 +168,26 @@ public function request_get_job( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Process a batch for the current job.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function request_process_job( $request ) {
+		$job = $this->get_job( $this->get_job_id_param( $request ) );
+
+		if ( $job && get_current_user_id() === $job->get_user_id() ) {
```

Comment:
Sorry, one more thing. We should probably check for this in the inverse and return `404` if `! $job || get_current_user_id() !== $job->get_user_id()` (similar to `request_get_job_logs` but with the user check).

Reply:
Just realized `get_job` already does the user check, ditched that from this and added the error in 816d73e